### PR TITLE
Don't setup try-except handler when Dynamo compiling

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133239

The reraise is not supported and so this just gunks up our actual exception handling. You can trigger this by hitting an exception inside of an NN module that has hooks on it. You end up graph breaking on the reraise here, and losing the inner stack trace from the actual exception that was raised.

This might be kind of controversial.  An alternate strategy is to support reraises in Dynamo or something but IDK this doesn't feel like the right place to apply force.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

File: torch/nn/modules/module.py [hunk 1]
```diff
@@ -1826,6 +1828,13 @@ def _call_impl(self, *args, **kwargs):
 
             return result
 
+        from torch.compiler import is_compiling
+
+        if is_compiling():
```
Comment (albanD) [inline torch/nn/modules/module.py:1833]: Can we have a comment here that explains why this is ok (what is explained in the PR description)?

File: torch/nn/modules/module.py [hunk 2]
```diff
@@ -1826,6 +1828,13 @@ def _call_impl(self, *args, **kwargs):
 
             return result
 
+        from torch.compiler import is_compiling
```
Comment (albanD) [inline torch/nn/modules/module.py:1831]: Shouldn't this be at the top level?

**Review by albanD — COMMENTED** [review]
Test?

File: torch/nn/modules/module.py (hunk 2 above)
Comment (ezyang) [reply torch/nn/modules/module.py:1831]: Worried about import loop lol

**Comment by ezyang** [comment]
Looking for more comments / opinions before I polish this more

**Comment by albanD** [comment]
Ho yeah, that sounds ok to me if we're saying that any error here during compilation leads to a complete bailout and re-running this in eager..
I can buy that Dynamo doesn't handle our cleanup logic well and needs help to smoothly bailout, it should but I agree not worth too much time.

**Review by anijain2305 — APPROVED** [review]
I am ok. If you are uncomfotable, I can look into implementing RERAISE as well. Let me know.

**Pull request closed without merging**